src/test/java/org/folio/services/invoice/InvoiceCancelServiceTest.java

```diff
@@ -96,81 +102,98 @@ public class InvoiceCancelServiceTest {
+  private RequestContext requestContext;
 
   @BeforeEach
   public void initMocks() {
-    requestContextMock = Mockito.mock(RequestContext.class, new RuntimeExceptionAnswer());
-    restClient = Mockito.mock(RestClient.class, new RuntimeExceptionAnswer());
-    doReturn(Vertx.vertx().getOrCreateContext()).when(requestContextMock).getContext();
+    MockitoAnnotations.openMocks(this);
+
+    Map<String, String> okapiHeaders = new HashMap<>();
+    okapiHeaders.put(OKAPI_URL, "http://localhost:" + mockPort);
+    okapiHeaders.put(X_OKAPI_TOKEN.getName(), X_OKAPI_TOKEN.getValue());
+    okapiHeaders.put(X_OKAPI_TENANT.getName(), X_OKAPI_TENANT.getValue());
+    okapiHeaders.put(X_OKAPI_USER_ID.getName(), X_OKAPI_USER_ID.getValue());
+    requestContext = new RequestContext(Vertx.vertx().getOrCreateContext(), okapiHeaders);
 
     BaseTransactionService baseTransactionService = new BaseTransactionService(restClient);
     OrderTransactionSummaryService orderTransactionSummaryService = new OrderTransactionSummaryService(restClient);
     EncumbranceService encumbranceService = new EncumbranceService(baseTransactionService, orderTransactionSummaryService);
     InvoiceTransactionSummaryService invoiceTransactionSummaryService = new InvoiceTransactionSummaryService(restClient);
-    VoucherRetrieveService voucherRetrieveService = new VoucherRetrieveService(restClient);
-    VoucherCommandService voucherCommandService = new VoucherCommandService(restClient, null,
-      voucherRetrieveService, null, null, null);
+    VoucherService voucherService = new VoucherService(restClient, new VoucherValidator());
     OrderLineService orderLineService = new OrderLineService(restClient);
     InvoiceLineService invoiceLineService = new InvoiceLineService(restClient);
     OrderService orderService = new OrderService(restClient, invoiceLineService, orderLineService);
     cancelService = new InvoiceCancelService(baseTransactionService, encumbranceService,
-      invoiceTransactionSummaryService, voucherCommandService, orderLineService,
+      invoiceTransactionSummaryService, voucherService, orderLineService,
       orderService);
   }
 
   @Test
-  public void cancelApprovedInvoiceTest() throws IOException {
+  public void cancelApprovedInvoiceTest(VertxTestContext vertxTestContext) throws IOException {
     Invoice invoice = getMockAs(APPROVED_INVOICE_SAMPLE_PATH, Invoice.class);
     List<InvoiceLine> invoiceLines = getMockAs(INVOICE_LINES_LIST_PATH, InvoiceLineCollection.class).getInvoiceLines();
 
     setupRestCalls(invoice, false);
 
-    CompletableFuture<Void> result = cancelService.cancelInvoice(invoice, invoiceLines, requestContextMock);
-    assertFalse(result.isCompletedExceptionally());
+    Future<Void> future = cancelService.cancelInvoice(invoice, invoiceLines, requestContext);
+    vertxTestContext.assertComplete(future)
+      .onSuccess(result -> {
+        vertxTestContext.completeNow();
+      })
+      .onFailure(vertxTestContext::failNow);
```

Comment:
I think checking the calls is important for this test. Also, we will need to do something similar for future tests, so we need to have a new solution. I suggest creating a new ticket to reimplement `verifyCalls()`, so that this ticket can move on.

Reply:
ok. i will try to restore it in scope of this story